src/main/java/ma/m3achaba/plantes/services/imp/AuthenticationUserService.java

```diff
@@ -0,0 +1,214 @@
+package ma.m3achaba.plantes.services.imp;
+
+
+import lombok.RequiredArgsConstructor;
+import ma.m3achaba.plantes.dto.RegisterRequest;
+import ma.m3achaba.plantes.model.User;
+import ma.m3achaba.plantes.repo.UserRepo;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationUserService {
+
+    private final PasswordEncoder passwordEncoder;
+    private final UserRepo user_repo;
+    private final JwtUtils jwtService;
+    private final AuthenticationManager authenticationManager;
+
+    public RegisterRequest register(RegisterRequest registerRequest) {
+        RegisterRequest response = new RegisterRequest();
+        try {
+            User user = new User();
```

Comment:
user builder

Reply:
in login/register/refresh data response is like this :
{
  "status": "success",
  "accessToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
  "expiresIn": 3600,
  "refreshToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
}UserResponse(id,nom,prenom,email,role) and UserRequest (nom,prenom,emailrole)
use Mapper interface ,creer mapper UserRequest to user  and user to UserResponse

Reply:
in controller same